01-upload-a-file/python/upload_sauce.py

```diff
@@ -0,0 +1,55 @@
+import os
+import sys
+import unittest
+from sauceclient import SauceClient
+from selenium import webdriver
+
+
+SAUCE_USERNAME = os.environ.get('SAUCE_USERNAME')
+SAUCE_ACCESS_KEY = os.environ.get('SAUCE_ACCESS_KEY')
+sauce = SauceClient(SAUCE_USERNAME, SAUCE_ACCESS_KEY)
+
+
+class UploadSauceTest(unittest.TestCase):
+    def setUp(self):
+        caps = webdriver.DesiredCapabilities.INTERNETEXPLORER
+
+        # massage to the correct state
+        caps['version'] = '8'
+        caps['platform'] = 'Windows XP'
+        caps['name'] = 'File Upload Test'
+
+        # http://saucelabs.com/docs/additional-config
+        caps['tags'] = ['elemental selenium', '01']
+
+        self.driver = webdriver.Remote(
+            desired_capabilities=caps,
+            command_executor="http://%s:%s@ondemand.saucelabs.com:80/wd/hub" % (SAUCE_USERNAME, SAUCE_ACCESS_KEY))
+
+        self.filename = 'some-file.txt'
+        self.full_filename = os.path.dirname(os.path.realpath(__file__)) + '/' + self.filename
```

Comment:
I find that importing as follows makes the code more concise, and thus a bit more readable:
``` python
from os.path import dirname, realpath

self.full_filename = dirname(realpath(__file__)) + '/' + self.filename
```
etc.

Personal preference of course!